### PR TITLE
830 : Add limit and metadata version params in v2 settings get end-point

### PR DESCRIPTION
**Pull request by rehammuzzamil**
Resolves https://github.com/opensrp/opensrp-server-web/issues/830

### CODE REVIEW

File: src/main/java/org/opensrp/web/rest/v2/SettingResource.java [hunk 1]
```diff
@@ -110,25 +114,40 @@ public void setSettingService(SettingService settingService, PhysicalLocationSer
...
 		String identifier = RestUtils.getStringFilter(SETTING_IDENTIFIER, request);
 		boolean resolveSettings = RestUtils.getBooleanFilter(AllConstants.Event.RESOLVE_SETTINGS, request);
+		String metadataVersion = RestUtils.getStringFilter(METADATA_VERSION, request);
+		String limit = RestUtils.getStringFilter(LIMIT, request);
 		Map<String, TreeNode<String, Location>> treeNodeHashMap = null;
 
 		if (StringUtils.isBlank(team) && StringUtils.isBlank(providerId) && StringUtils.isBlank(locationId)
-				&& StringUtils.isBlank(teamId) && StringUtils.isBlank(team) && StringUtils.isBlank(serverVersion)) {
+				&& StringUtils.isBlank(teamId) && StringUtils.isBlank(team) && StringUtils.isBlank(serverVersion)
+				&& StringUtils.isBlank(metadataVersion)) {
```
Comment (bennsimon) [inline src/main/java/org/opensrp/web/rest/v2/SettingResource.java:123]: Wont this break the current functionality `StringUtils.isBlank(metadataVersion)`?

**Review by bennsimon — CHANGES_REQUESTED** [review]

File: src/main/java/org/opensrp/web/rest/v2/SettingResource.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/v2/SettingResource.java:123]: This check is to ensure that all parameters can not be null. Hence, added meta version field here as well. @bennsimon 
Please let me know if there is anything incorrect. 
settingsv2get

Comment (bennsimon) [reply src/main/java/org/opensrp/web/rest/v2/SettingResource.java:123]: This is a new parameter so for the devs that were supplying the other params only it should work.

Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/v2/SettingResource.java:123]: Yes, `metadata_version` and `limit` request params are not required fields. This check is to ensure that all request parameters can not be null, we need to supply at least a single request param from the given list. Thus added the check. 
Should I remove validation from here? @bennsimon

Comment (bennsimon) [reply src/main/java/org/opensrp/web/rest/v2/SettingResource.java:123]: leave it

**Review by bennsimon — APPROVED** [review]

**Review by bennsimon — APPROVED** [review]

**Pull request merged**